### PR TITLE
ROE-1729 updating public register validation and messages

### PR DESCRIPTION
**Pull request by SVarallo**

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidatorTest.java [hunk 1]
```diff
@@ -433,7 +433,7 @@ void testErrorReportedWhenRegistrationNumberFieldContainsInvalidCharacters() {
     @Test
     void testNoErrorReportedWhenPublicRegisterNameAndJurisdictionFieldIsMaxLength() {
         entityDto.setOnRegisterInCountryFormedIn(true);
-        entityDto.setPublicRegisterName(StringUtils.repeat("A", 79));
+        entityDto.setPublicRegisterName(StringUtils.repeat("A", 78));
```
Comment (mattch1) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidatorTest.java:436]: Could we run this with 79 characters as this would be the edge case

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java [hunk 2]
```diff
@@ -121,7 +121,7 @@ private boolean validatePublicRegisterName(String publicRegisterName, String pub
         var compoundQualifiedFieldName = String.format("%s and %s", qualifiedFieldNamePublicRegisterName, qualifiedFieldNamePublicRegisterJurisdiction);
 
         return StringValidators.isNotBlank(publicRegisterName, qualifiedFieldNamePublicRegisterName, errors, loggingContext)
-                && StringValidators.isLessThanOrEqualToMaxLength(String.format(CONCATENATED_STRING_FORMAT, publicRegisterName, publicRegisterJurisdiction), 160, compoundQualifiedFieldName, errors, loggingContext)
+                && StringValidators.isLessThanOrEqualToMaxLength(String.format(CONCATENATED_STRING_FORMAT, publicRegisterName, publicRegisterJurisdiction), 159, compoundQualifiedFieldName, errors, loggingContext)
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java:124]: I think the length check should still be 160, as the name and jurisdiction should be max 159, the comma between them makes it 160.
So i believe we just need to change the error message that gets returned to tell the user they can only enter 159 chars. I would check this with Sarah (who raised the bug)

Comment (sarahocarroll) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java:124]: Yes, that's correct - the validation message should read that they are only allowed to enter a max of 159 characters.

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidatorTest.java (hunk 1 above)
Comment (SVarallo) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidatorTest.java:436]: But wouldn't this move it up to 160 when you include the comma?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java (hunk 2 above)
Comment (SVarallo) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java:124]: The error message is generated by the validation itself right? Would that require extra work?

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidatorTest.java (hunk 1 above)
Comment (mattch1) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidatorTest.java:436]: yep, thats the edge case as long as it's not >=161 or > 160 it should pass.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java [hunk 3]
```diff
@@ -131,7 +131,7 @@ private boolean validatePublicRegisterJurisdiction(String publicRegisterName, St
         var compoundQualifiedFieldName = String.format("%s and %s", qualifiedFieldNamePublicRegisterName, qualifiedFieldNamePublicRegisterJurisdiction);
 
         return StringValidators.isNotBlank(publicRegisterJurisdiction, qualifiedFieldNamePublicRegisterJurisdiction, errors, loggingContext)
-                && StringValidators.isLessThanOrEqualToMaxLength(String.format(CONCATENATED_STRING_FORMAT, publicRegisterName, publicRegisterJurisdiction), 160, compoundQualifiedFieldName, errors, loggingContext)
+                && StringValidators.isLessThanOrEqualToMaxLength(String.format(CONCATENATED_STRING_FORMAT, publicRegisterName, publicRegisterJurisdiction), 159, compoundQualifiedFieldName, errors, loggingContext)
```
Comment (mattch1) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java:134]: Just had a look. The concatenated string it supposed to be 160 or less, i.e. with the comma, which means the user input should be 159 to account for it, so I'm not sure this should change as we are testing the string as whole. If we want to test user input only then we'd need to remove the concatenated string format, append the two string and then set it to 159. Any error messages both in the api and the web should specify "user input" when stating 159. @markpit and @sarahocarroll what do you think?

Comment (markpit) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java:134]: i'll pull the branch and have a look, i was hoping it would just be a case of updating the error message text.

Comment (markpit) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/EntityDtoValidator.java:134]: Yes, what Matt said should work. instead of using the CONCATENATED_STRING_FORMAT to join the 2 strings together, you could just create a new constant - CONCATENATED_STRING_FORMAT_NO_COMMA = %s%s
That will then check the length of the 2 strings together and use the 159 value like you have done.

**Review by mattch1 — APPROVED** [review]
